sql/sql_select.cc

```diff
@@ -25007,6 +25008,25 @@ int setup_order(THD *thd, Ref_ptr_array ref_pointer_array, TABLE_LIST *tables,
 }
 
 
+// Relies on RAII to invoke a work function on calling scope's
+// exit.  Typically used when a scope has multiple return points
+// and some cleanup must be done in all cases.  Prevents
+// calling code from missing one such cleanup invocation.
+class OnScopeExit
+{
+  std::function<void()> _work;
+
+public:
+  OnScopeExit(std::function<void()> work)
+    : _work(work)
+  {}
+
+  ~OnScopeExit()
+  {
+    _work();
+  }
+};
+
```

Comment:
No, don't do that. You can use `SCOPE_EXIT` from `include/scope.h`

Reply:
I missed `scope_exit`, thank you; it does exactly what I need 👍

Reply:
uppercase. Like
```c++
SCOPE_EXIT([order] () {
   ...
});
```